Fix `Delegate` dispatch.

AaronRobinsonMSFT opened the pull request:
/cc @janvorli @cshung

AaronRobinsonMSFT committed:
Fix Delegate dispatch.

huoyaoyuan commented on src/coreclr/vm/comdelegate.cpp:
```diff
@@ -1901,16 +1905,21 @@ Stub* COMDelegate::GetInvokeMethodStub(EEImplMethodDesc* pMD)
+        pCode->EmitLDFLD(FIELD__DELEGATE__METHOD_PTR);
+
+        mdToken sigTok = pCode->GetSigToken(pSig, cbSig);
+        pCode->EmitCALLI(sigTok, sig.NumFixedArgs(), fReturnVal);
```
Is there any concern about GC safety of collectible methods? (dotnet/runtime#105082)

I'm not familiar about how interpreter interacts with GC.

AaronRobinsonMSFT replied:
It really shouldn't impact the interpreter scenario, but it costs us nothing to make it correct. This is throw away code right now, but I will make it correct. Thanks for pointing this out.

huoyaoyuan replied:
I originally used the same mechanism in dotnet/runtime#104731 . You may keep the mentioned points in mind when bringing the interpreter for production quality. /cc @jkotas

jkotas replied:
This is throw-away code. We do not plan to bring this implementation of the interpreter to quality. The change looks good as a temporary measure. It has multiple corner-case problems and that's fine. 

The only 100% correct fix would be to treat `Invoke` as an intrinsic in the interpreter (ie keep this as is and change the code in interpreter.cpp instead).

AaronRobinsonMSFT committed:
Update Delegate impl for interpreter.
Add GetMethodTable intrinsic.
Fix issue with RW/RX confusion with unwinder headers.
